ClimsoftVer4/ClimsoftVer4/ucrDay.vb

```diff
@@ -1,22 +1,25 @@
 ﻿Public Class ucrDay
     'Private strDaysTableName As String = "days"
     Private strDay As String = "day"
-    Private dtb28 As New DataTable
-    Private dtb29 As New DataTable
-    Private dtb30 As New DataTable
-    Private dtb31 As New DataTable
+    Private dtb28 As DataTable
+    Private dtb29 As DataTable
+    Private dtb30 As DataTable
+    Private dtb31 As DataTable
     Private ucrLinkedYear As ucrYearSelector
     Private ucrLinkedMonth As ucrMonth
 
 
     Public Sub InitialiseControl()
-        'MyBase.PopulateControl()
-        dtbRecords = New DataTable
-        dtbRecords.Columns.Add(strDay, GetType(Integer))
-        'dtb31.Columns.Add(strDay, GetType(Integer))
-        'dtb30.Columns.Add(strDay, GetType(Integer))
-        'dtb29.Columns.Add(strDay, GetType(Integer))
-        'dtb28.Columns.Add(strDay, GetType(Integer))
+        'MyBase.InitialiseControl()
+        dtb31 = New DataTable
+        dtb30 = New DataTable
+        dtb29 = New DataTable
+        dtb28 = New DataTable
+
+        dtb31.Columns.Add(strDay, GetType(Integer))
+        dtb30.Columns.Add(strDay, GetType(Integer))
+        dtb29.Columns.Add(strDay, GetType(Integer))
+        dtb28.Columns.Add(strDay, GetType(Integer))
 
```

Comment:
@shadrackkibet just remembered that the above could be changed as follows depending on your preference. They both work the same way.

From

```
dtb31.Columns.Add(strDay, GetType(Integer))
 dtb30.Columns.Add(strDay, GetType(Integer))
dtb29.Columns.Add(strDay, GetType(Integer))
dtb28.Columns.Add(strDay, GetType(Integer))
```

to

```
 Dim col As New DataColumn(strDay, GetType(Integer))
        dtb31.Columns.Add(col)
        dtb30.Columns.Add(col)
        dtb29.Columns.Add(col)
        dtb28.Columns.Add(col)
```